crates/polars-core/src/fmt.rs

```diff
@@ -949,12 +949,32 @@ impl Series {
                 self.get(1).unwrap(),
                 self.get(2).unwrap()
             ),
-            _ => format!(
-                "[{}, {}, … {}]",
-                self.get(0).unwrap(),
-                self.get(1).unwrap(),
-                self.get(self.len() - 1).unwrap()
-            ),
+            _ => {
+                let max_items = std::env::var(FMT_TABLE_CELL_LIST_LEN)
+                    .as_deref()
+                    .unwrap_or("")
+                    .parse()
+                    .map_or(3, |n: i64| if n < 0 { self.len() } else { n as usize });
+
+                let mut result = "[".to_owned();
+
+                for (i, item) in self.iter().enumerate() {
+                    result.push_str(&item.to_string());
+
+                    if i != self.len() - 1 {
+                        result.push_str(", ");
+                    }
+
+                    if i == max_items - 2 {
+                        result.push_str("… ");
+                        result.push_str(&self.get(self.len() - 1).unwrap().to_string());
```

Comment:
I think we should use a `write!` macro here, to elide the superfluous `to_string` call.

Reply:
for sure :smile: